xapian-core/include/xapian/query.h

```diff
@@ -375,6 +375,22 @@ class XAPIAN_VISIBILITY_DEFAULT Query {
 	  Xapian::termcount wqf = 1,
 	  Xapian::termpos pos = 0);
 
+    /** Construct a Query object for a term and weighting scheme.
+     *
+     *  @param term The term.  An empty string constructs a query matching
+     *		    all documents (@a MatchAll is a static instance of this).
+     *  @param wt   Weighting scheme. Use the supplied weighing scheme
+     *  	    to calculate weight for term.
+     *  @param wqf  The within-query frequency. (default: 1)
+     *  @param pos  The query position.  Currently this is mainly used to
+     *		    determine the order of terms obtained via
+     *		    get_terms_begin(). (default: 0)
+     */
+    Query(const std::string& term,
+	  const Xapian::Weight*  wt,
+	  Xapian::termcount wqf = 1,
+	  Xapian::termpos pos = 0);
```

Comment:
We probably need to update the Query Description of QueryTerm::get_description to show we added an external weight

I am probably thinking of something like this:
```
+    if (weight != 0) {
+       desc  += "^coord";
+    }

```